src/routes/subject-router.ts

```diff
@@ -1,6 +1,6 @@
-import express from "express";
+import express, { response } from "express";
 import { Subject } from "../models/subjects";
-import { GenericSubjectService } from "../utils/generic-service-initializer";
+import { GenericSubjectService, SubjectsService } from "../utils/generic-service-initializer";
```

Comment:
you have SubjectService initialized in generic-service-initializer?
Initialize it in the place, instead. The generics are used in multiple places, that's why they have a special initializer, not needed for specific ones